provisioning/service/samples/bulkupdate_sample.js

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+var assert = require('chai').assert;
+
+var provisioningClient = require('azure-iot-provisioning-service-client');
+
+var serviceClient = provisioningClient.DeviceEnrollment.fromConnectionString(process.argv[2]);
+
+var enrollment1 = {
+  registrationId: 'first',
+  provisioningStatus: 'disabled',
+  attestation: {
+    type: 'tpm',
+    tpm: {
+      endorsementKey: 'a',
+      storageRootKey: 'b'
+    }
+  }
+};
+
+var enrollment2 = {
+  registrationId: 'second',
+  provisioningStatus: 'disabled',
+  attestation: {
+    type: 'tpm',
+    tpm: {
+      endorsementKey: 'a',
+      storageRootKey: 'b'
+    }
+  }
+};
+
+var enrollment3 = {
+  registrationId: 'third',
+  provisioningStatus: 'disabled',
+  attestation: {
+    type: 'tpm'
+  }
+};
+
+var badCreateBulkOperation = {
+  mode: 'create',
+  enrollments: []
+};
+badCreateBulkOperation.enrollments.push(enrollment1);
+badCreateBulkOperation.enrollments.push(enrollment2);
+badCreateBulkOperation.enrollments.push(enrollment2);
+
+//
+// This bulk operation will fail.  The reason it fails is because we have
+// a duplicate entry in the bulk array.  This will NOT produce a BulkOperationResult.
+//
+serviceClient.bulkOperation(badCreateBulkOperation, function(err, result, response) {
+  if (err) {
+    console.log('The bulk operation result is: ' + err);
+  }
+  console.log('The value of the result operation should be undefined: ',JSON.stringify(result, null, 2));
+  //
+  // Get rid of the duplicate entry and put in an entry without an actual attestation.
+  // Some of the bulk create will work but we should get a bulk operation result
+  // that indicates the third one failed.
+  //
```

Comment:
why show things that fail? this is making the sample unnecessarily hard to read. as the reader, i'm wondering why you're messing with me right now.

Reply:
When actually working with the bulkOperation function I found it's error indication a bit confusing at first.  I wanted to provide examples to a developer on the two different ways that errors could be indicated to the developer.

The first error was the pre-condition error.  In that case there would be no results and no enrollments would have been created at all.

In the second case there was an error with one of the enrollments.  In that case the err parameter is null and it's the bulkResults that contains the error information.

I thought that this would be helpful to the developer.

---
In reply to: [147066400](https://github.com/Azure/azure-iot-sdk-node/pull/121#discussion_r147066400) [](ancestors = 147066400)

Reply:
that's a lot nicer. thanks!

---
In reply to: [147200379](https://github.com/Azure/azure-iot-sdk-node/pull/121#discussion_r147200379) [](ancestors = 147200379,147066400)